src/rqa.jl

```diff
@@ -410,29 +434,33 @@ Using this function is much more efficient than calling all individual functions
 one by one.
 
 ## Return
-The returned value is a NamedTuple with the following entries:
-
-* `RR`: recurrence rate (see [`recurrencerate`](@ref))
-* `TRANS`: transitivity (see [`transitivity`](@ref))
-* `DET`: determinsm (see [`determinism`](@ref))
-* `L`: average length of diagonal structures (see [`dl_average`](@ref))
-* `Lmax`: maximum length of diagonal structures (see [`dl_max`](@ref))
-* `DIV`: divergence (see [`divergence`](@ref))
-* `ENTR`: entropy of diagonal structures (see [`dl_entropy`](@ref))
-* `TREND`: trend of recurrences (see [`trend`](@ref))
-* `LAM`: laminarity (see [`laminarity`](@ref))
-* `TT`: trapping time (see [`trappingtime`](@ref))
-* `Vmax`: maximum length of vertical structures (see [`vl_max`](@ref))
-* `VENTR`: entropy of vertical structures (see [`vl_entropy`](@ref))
-* `MRT`: mean recurrence time (see [`meanrecurrencetime`](@ref))
-* `RTE` recurrence time entropy (see [`rt_entropy`](@ref))
-* `NMPRT`: number of the most probable recurrence time (see [`nmprt`](@ref))
-
+The returned value contains the following entries,
+which can be retrieved as from a dictionary (e.g. `results[:RR]`, etc.):
+
+* `:RR`: recurrence rate (see [`recurrencerate`](@ref))
+* `:TRANS`: transitivity (see [`transitivity`](@ref))
+* `:DET`: determinsm (see [`determinism`](@ref))
+* `:L`: average length of diagonal structures (see [`dl_average`](@ref))
+* `:Lmax`: maximum length of diagonal structures (see [`dl_max`](@ref))
+* `:DIV`: divergence (see [`divergence`](@ref))
+* `:ENTR`: entropy of diagonal structures (see [`dl_entropy`](@ref))
+* `:TREND`: trend of recurrences (see [`trend`](@ref))
+* `:LAM`: laminarity (see [`laminarity`](@ref))
+* `:TT`: trapping time (see [`trappingtime`](@ref))
+* `:Vmax`: maximum length of vertical structures (see [`vl_max`](@ref))
+* `:VENTR`: entropy of vertical structures (see [`vl_entropy`](@ref))
+* `:MRT`: mean recurrence time (see [`meanrecurrencetime`](@ref))
+* `:RTE` recurrence time entropy (see [`rt_entropy`](@ref))
+* `:NMPRT`: number of the most probable recurrence time (see [`nmprt`](@ref))
+
+All the parameters returned by `rqa` are `Float64` numbers,
+even for parameters like `:Lmax`, `:Vmax` or `:NMPRT` which are integer values.
 In the case of empty histograms (e.g. no existing vertical lines
 less than the keyword `lminvert`) the average and maximum values
-(`L`, `Lmax`, `TT`, `Vmax`, `MRT`)
-are returned as `0.0` but their respective entropies (`ENTR`, `VENTR`, `RTE`)
+(`:L`, `:Lmax`, `:TT`, `:Vmax`, `:MRT`)
+are returned as `0.0` but their respective entropies (`:ENTR`, `:VENTR`, `:RTE`)
 are returned as `NaN`.
+`:TRANS` is only returned when the input is a square matrix.
```

Comment:
Now that I read this, isn't it more consistent to return `NaN` for TRANS of non-square, similarly with how the entropies are NaN when no other sensible value is in order?

Reply:
The situation of TRANS is not exactly the same as that of the entropies. Entropies return `NaN` when there are no lines (longer than the minum length) in the matrix, which is a "legitimate" case. (The formula of the Shannon entropy (`-sum(x .* log.(x))`) would give zero for an empty histogram, but we coerce `NaN` in order to distinguish that case from that where all the lines have the same length, which also returns zero.) On the other hand, there is no point in trying to calculate TRANS for non-square matrices (its definition is related to a single set of vertices, so the matrix must have the same number of columns and rows).

Actually, I think that the `transitivity` function could (should?) throw an error (with the same informative message) instead of a warning, when it is called with non-square matrices.

Reply:
if it throws an error what happens with calling `rqa` for non-square matrices?

Reply:
Nothing happens: since dictionaries are mutable, the dictionary is initially created without the `:TRANS` key, and it is added only if the matrix is square.

Reply:
a okay sure.